path_role_test.go

```diff
@@ -38,624 +38,699 @@ func getBackend(t *testing.T) (logical.Backend, logical.Storage) {
-		"user_claim":      "user",
-		"groups_claim":    "groups",
-		"bound_cidrs":     "127.0.0.1/8",
-		"policies":        "test",
-		"period":          "3s",
-		"ttl":             "1s",
-		"num_uses":        12,
-		"max_ttl":         "5s",
-	}
-
-	expectedSockAddr, err := sockaddr.NewSockAddr("127.0.0.1/8")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	expected := &jwtRole{
-		TokenParams: tokenutil.TokenParams{
-			TokenPolicies:   []string{"test"},
-			TokenPeriod:     3 * time.Second,
-			TokenTTL:        1 * time.Second,
-			TokenMaxTTL:     5 * time.Second,
-			TokenNumUses:    12,
-			TokenBoundCIDRs: []*sockaddr.SockAddrMarshaler{{SockAddr: expectedSockAddr}},
-		},
-		RoleType:            "jwt",
-		Policies:            []string{"test"},
-		Period:              3 * time.Second,
-		BoundSubject:        "testsub",
-		BoundAudiences:      []string{"vault"},
-		BoundClaimsType:     "string",
-		UserClaim:           "user",
-		GroupsClaim:         "groups",
-		TTL:                 1 * time.Second,
-		MaxTTL:              5 * time.Second,
-		ExpirationLeeway:    0,
-		NotBeforeLeeway:     0,
-		ClockSkewLeeway:     0,
-		NumUses:             12,
-		BoundCIDRs:          []*sockaddr.SockAddrMarshaler{{SockAddr: expectedSockAddr}},
-		AllowedRedirectURIs: []string(nil),
-	}
-
-	req := &logical.Request{
-		Operation: logical.CreateOperation,
-		Path:      "role/plugin-test",
-		Storage:   storage,
-		Data:      data,
-	}
-
-	resp, err := b.HandleRequest(context.Background(), req)
-	if err != nil || (resp != nil && resp.IsError()) {
-		t.Fatalf("err:%s resp:%#v\n", err, resp)
-	}
-	actual, err := b.(*jwtAuthBackend).role(context.Background(), storage, "plugin-test")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if !reflect.DeepEqual(expected, actual) {
-		t.Fatalf("Unexpected role data: expected %#v\n got %#v\n", expected, actual)
-	}
-
-	// Test no user claim
-	data = map[string]interface{}{
-		"policies": "test",
-	}
-
-	req = &logical.Request{
-		Operation: logical.CreateOperation,
-		Path:      "role/test2",
-		Storage:   storage,
-		Data:      data,
-	}
-
-	resp, err = b.HandleRequest(context.Background(), req)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if resp != nil && !resp.IsError() {
-		t.Fatalf("expected error")
-	}
-	if resp.Error().Error() != "a user claim must be defined on the role" {
-		t.Fatalf("unexpected err: %v", resp)
-	}
-
-	// Test no binding
-	data = map[string]interface{}{
-		"role_type":  "jwt",
-		"user_claim": "user",
-		"policies":   "test",
-	}
-
-	req = &logical.Request{
-		Operation: logical.CreateOperation,
-		Path:      "role/test3",
-		Storage:   storage,
-		Data:      data,
-	}
-
-	resp, err = b.HandleRequest(context.Background(), req)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if resp != nil && !resp.IsError() {
-		t.Fatalf("expected error")
-	}
-	if !strings.HasPrefix(resp.Error().Error(), "must have at least one bound constraint") {
-		t.Fatalf("unexpected err: %v", resp)
-	}
+	t.Run("", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("sunny path", func(t *testing.T) {
```

Reply:
Would that be a good name for this one? Feel free to change it if that doesn't really fit.

Reply:
How about `"happy path"`?